programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -524,25 +408,29 @@ fn translate_string_and_do(
     }
 }
 
-type SyscallContext<'a, 'b> = Rc<RefCell<&'a mut InvokeContext<'b>>>;
-
 #[macro_export]
 macro_rules! declare_syscall {
-    ($(#[$attr:meta])* $name:ident, $call:item) => {
+    ($(#[$attr:meta])* $name:ident, $inner_call:item) => {
         $(#[$attr])*
-        pub struct $name<'a, 'b> {
-            pub(crate) invoke_context: SyscallContext<'a, 'b>,
-        }
-        impl<'a, 'b> $name<'a, 'b> {
-            pub fn init(
-                invoke_context: SyscallContext<'a, 'b>,
-            ) -> Box<(dyn SyscallObject<BpfError> + 'a)> {
-                Box::new(Self { invoke_context })
+        pub struct $name {}
+        impl $name {
+            $inner_call
+            pub fn call(
+                invoke_context: &mut InvokeContext,
+                arg_a: u64,
+                arg_b: u64,
+                arg_c: u64,
+                arg_d: u64,
+                arg_e: u64,
+                memory_mapping: &mut MemoryMapping,
+                result: &mut ProgramResult,
+            ) {
+                let converted_result: ProgramResult = Self::inner_call(
+                    invoke_context, arg_a, arg_b, arg_c, arg_d, arg_e, memory_mapping,
+                ).into();
+                *result = converted_result;
```

Comment:
nit: I'd expand `$inner_call` here, as a nested function. That way you can still call it `fn call(...)` in macro instantiations instead of having to call it `fn inner_call(...)` (there are other ways to do this, but this is the easiest)

Reply:
If I understand correctly, this is purely cosmetic. So, would be a separate PR later.